content/general/concepts/search-algorithms/search-algorithms.md

```diff
@@ -0,0 +1,64 @@
+---
+Title: 'Search Algorithms'
+Description: 'search algorithms are a class of algorithms used to find an optimal solution to a problem by exploring a search space.'
+Subjects:
+  - 'Machine Learning'
+  - 'Computer Science'
+Tags:
+  - 'AI'
+  - 'Algorithms'
+  - 'Graphs'
+  - 'Graph Search'
+CatalogContent:
+  - ''
+---
+
+A **Search algorithm** is a type of algorithm used in artificial intelligence to find the best or most optimal solution to a problem by exploring a set of possible solutions, also called a **search space**. Essentially, search algorithms help computers search through a large number of possiblities in order to find a solution that works the best, given a set of constraints.
+
+Search Algorithms typically operate by organizing the search space into a particular type of graph, commonly a tree, and evaluate the best score, or **cost**, of traversing each branch of the tree. The algorithm produces a path from the start state to the goal state with the best cost that the specific type of search algorithm being used can find.
+
+Search algorithms are typically organized into two categories:
+
+### Uninformed Search
+
+Uninformed search algorithms consist of general purpose traverse the state space or search tree without any information about how good a state is. These are also referred to as blind search algorihtms.
+
+### Informed Search
+
+Informed search algorithms have information about the goal during the traversal. This allows the search to prioritize its expansion toward the goal state instead of exploring directions that may have a good cost but dont lead to the goal. By including extra rules that help us estimate how close we are to the goal, known as _heuristics_, Informed search algorithms can be more computationally efficient when searching for a path to the goal state.
```

Comment:
```suggestion
Informed search algorithms have information about the goal during the traversal. This allows the search to prioritize its expansion toward the goal state instead of exploring directions that may yield a favorable cost but don't lead to the goal, or global optimum. By including extra rules that aid in estimating the location of the goal, known as _heuristics_, informed search algorithms can be more computationally efficient when searching for a path to the goal state.
```